pkg/cache/tas_flavor_snapshot.go

```diff
@@ -296,19 +297,34 @@ func (s *TASFlavorSnapshot) updateCountsToMinimum(domains []*domain, count int32
 }
 
 func (s *TASFlavorSnapshot) buildAssignment(domains []*domain) *kueue.TopologyAssignment {
-	assignment := kueue.TopologyAssignment{
-		Levels:  s.levelKeys,
-		Domains: make([]kueue.TopologyDomainAssignment, 0),
-	}
+	assignment := kueue.TopologyAssignment{}
 	// lex sort domains
 	slices.SortFunc(domains, func(a, b *domain) int {
 		return cmp.Compare(a.id, b.id)
 	})
-	for _, domain := range domains {
-		assignment.Domains = append(assignment.Domains, kueue.TopologyDomainAssignment{
-			Values: domain.levelValues,
-			Count:  domain.state,
-		})
+	// assign only hostname values if topology defines it
+	if idx, found := s.resolveLevelIdx(corev1.LabelHostname); found {
```

Comment:
I would suggest to check specifically the last level. Maybe we can add validation that this label is the lowest in the CRD. This would be also consistent with what I do here: https://github.com/kubernetes-sigs/kueue/pull/3678/files#diff-5414dcb324187d1d853329b419691cde79632a6d77e1a7c2b503dc1a3688c081R145-R146

Reply:
Done